Create rule S6316: Kotlin coroutines api for timeouts should be used (#182)
SonarSource/rspec: rules/S6316/kotlin/rule.adoc

> Sometimes there is a need to cancel the execution of the coroutine after some time. You can do it manualy using `delay()` and `cancel()` functions. But this way is verbose and errorprone. There is an easier way to manage timeouts by using functions `withTimeout()` or `withTimeoutOrNull()`.

Reviewer: ```suggestion
Sometimes there is a need to cancel the execution of a coroutine after a given period of time. You can do this manually by combining the `delay()` and `cancel()` functions. However, this technique is verbose and error-prone. The easier way to manage timeouts is using the function `withTimeout()` or `withTimeoutOrNull()`.
```

> Sometimes there is a need to cancel the execution of a coroutine after a given period of time. You can do this manually by combining the `delay()` and `cancel()` functions. However, this technique is verbose and error-prone. The easier way to manage timeouts is using the function `withTimeout()` or `withTimeoutOrNull()`.

Reviewer: ```suggestion
Sometimes there is the need to cancel the execution of a coroutine after a given period of time. You can do this manually by combining the `delay()` and `cancel()` functions. However, this technique is verbose and error-prone. An easier way to manage timeouts is using the function `withTimeout()` or `withTimeoutOrNull()`.
```

> `withTimeout` function will throw a `TimeoutCancellationException` in case timeout was reached, while `withTimeoutOrNull` will just return null in this case.

Reviewer: ```suggestion
The `withTimeout` function will throw a `TimeoutCancellationException` when the timeout is reached, while `withTimeoutOrNull` will simply return `null` instead.
```

> This rule raises an issue if timeouts mechanism is implemented manually instead of using appropriate functions.

Reviewer: ```suggestion
This rule raises an issue if timeout mechanisms are implemented manually instead of using appropriate built-in functions.
```
